de.dlr.sc.virsat.model.extension.mechanical/src/de/dlr/sc/virsat/model/extension/mechanical/catia/CatiaImporter.java

```diff
@@ -9,33 +9,85 @@
  *******************************************************************************/
 package de.dlr.sc.virsat.model.extension.mechanical.catia;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.common.command.CompoundCommand;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.edit.command.AddCommand;
+import org.eclipse.emf.edit.domain.EditingDomain;
+
 import com.github.cliftonlabs.json_simple.JsonObject;
 
+import de.dlr.sc.virsat.model.concept.types.structural.BeanStructuralElementInstance;
 import de.dlr.sc.virsat.model.concept.types.structural.IBeanStructuralElementInstance;
+import de.dlr.sc.virsat.model.dvlm.Repository;
+import de.dlr.sc.virsat.model.dvlm.categories.CategoriesPackage;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.concepts.util.ActiveConceptHelper;
 import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
 import de.dlr.sc.virsat.model.extension.mechanical.catia.util.CatiaHelper;
+import de.dlr.sc.virsat.model.extension.visualisation.Activator;
+import de.dlr.sc.virsat.model.extension.visualisation.model.Visualisation;
+import de.dlr.sc.virsat.project.editingDomain.VirSatEditingDomainRegistry;
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+import de.dlr.sc.virsat.project.structure.VirSatProjectCommons;
 
 /**
- * This class imports a JSON representation of a product tree
+ * This class imports a JSON representation of system model
  *
  */
 public class CatiaImporter {
 
+	private EditingDomain editingDomain;
+
 	/**
 	 * Main method that imports a JSON representation of our system and updates the
 	 * model accordingly
 	 * 
+	 * @param editingDomain
+	 *            the editing domain of the current project
 	 * @param jsonObject
 	 *            the JSON Object
 	 * @param mapJsonUuidToSEI
 	 *            the mapping of JSON element IDs to the existing trees
+	 * 
+	 * @return the emf command to execute the import
 	 */
-	public void transform(JsonObject jsonObject, Map<String, StructuralElementInstance> mapJsonUuidToSEI) {
+	public Command transform(EditingDomain editingDomain, JsonObject jsonObject,
+			Map<String, StructuralElementInstance> mapJsonUuidToSEI) {
+
+		CompoundCommand importCommand = new CompoundCommand();
+		this.editingDomain = editingDomain;
+
+		boolean worked = true;
+
+		// Import parts
+		for (JsonObject part : CatiaHelper.getListOfAllJSONParts(jsonObject)) {

```

Comment:
If `mapJsonUuidToSEI` is not used for filtering SEIs to update, than what's the point of passing it from outside? We could also create it directly in this method

Reply:
Well, it is intended to be used outside. I just wrote a test case for that. Here I just missed filtering with the map....